### PR TITLE
meshreg: nacos source support security access

### PR DESCRIPTION
**Pull request by believening**
meshregistry nacos source add new configuration `Username` and `Password` to access nacos server using authentication.

usage:

```
apiVersion: v1
data:
  cfg: |
    bundle:
      modules:
      - name: meshregistry
        kind: meshregistry
    enable: true
  cfg_meshregistry: |
    name: meshregistry
    kind: meshregistry
    enable: true
    mode: BundleItem
    general:
      LEGACY:
        MeshConfigFile: ""
        RevCrds: ""
        Mcp:
          EnableIncPush: false
        NacosSource:
          Enabled: true
          Address:
          - "http://nacos-0.nacos-headless.default:8848"
          - "http://nacos-1.nacos-headless.default:8848"
          - "http://nacos-2.nacos-headless.default:8848"
          Mode: polling
          Username: nacos
          Password: nacos
```

### CODE REVIEW

File: staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go [hunk 1]
```diff
@@ -121,3 +152,48 @@ func (c *client) Instances() ([]*instanceResp, error) {
...
+func (c *client) autoRefresh() {
+	if c.username == "" || c.password == "" {
+		return
+	}
+	go func() {
+		timer := time.NewTimer(time.Duration((c.tokenTTL - c.tokenTTL/10)) * time.Second)
+		for {
+			<-timer.C
+			c.login()
+			timer.Reset(time.Duration((c.tokenTTL - c.tokenTTL/10)) * time.Second)
```
Comment (YonkaFang) [inline staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go:196]: even reset timer when login fails?

File: staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go [hunk 2]
```diff
@@ -121,3 +152,48 @@ func (c *client) Instances() ([]*instanceResp, error) {
...
 	return instanceAll, nil
 }
+
+func (c *client) login() {
+	if c.username == "" || c.password == "" {
+		return
+	}
+	url := c.chooseURL() + loginPath
+	body := strings.NewReader("username=" + c.username + "&password=" + c.password)
+	resp, err := c.call(http.MethodPost, url, body)
```
Comment (YonkaFang) [inline staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go:162]: i'm not sure if server will revoke the previous token when we re-login and if that the current access may get 403 err.

File: staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go (hunk 1 above)
Comment (believening) [reply staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go:204]: reset ttl as default when login fails

File: staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go [hunk 3]
```diff
@@ -121,3 +152,56 @@ func (c *client) Instances() ([]*instanceResp, error) {
...
+		return
+	}
+	needResetTTL := false
+	defer func() {
+		if needResetTTL {
+			c.tokenTTL = defaultNacosTokenTTL
+		}
+	}()
+	url := c.chooseURL() + loginPath
+	body := strings.NewReader("username=" + c.username + "&password=" + c.password)
```
Comment (YonkaFang) [inline staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go:167]: urlencode？

Comment (YonkaFang) [reply staging/src/slime.io/slime/modules/meshregistry/pkg/source/nacos/httpclient.go:167]: peopel usually use special chars in password string

**Pull request merged**